### PR TITLE
Queue Supplemental Claims Form4142 generation and sending

### PR DESCRIPTION
**Pull request by kylesoskin**
## Summary

Adds a sidekiq job for form4142 submissions for supplemental claims. 

Previously the form pdf was generated and sent synchronously at the controller layer. This holds up the request from returning to the user. Making it a sidekiq job allows it to return right away to the user and also offers the ability to enable it to retry in case it fails. 

## Related issue(s)
- https://github.com/department-of-veterans-affairs/va.gov-team/issues/61012

## Testing done

- Local
- Rspec test added. 

## Screenshots
_Note: Optional_

## What areas of the site does it impact?
*(Describe what parts of the site are impacted and*if*code touched other areas)*

## Acceptance criteria

- [x]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x]  No error nor warning in the console.
- [x]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [x]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog or Grafana (if applicable)
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature

## Requested Feedback

(OPTIONAL)_What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

File: app/controllers/v1/supplemental_claims_controller.rb [hunk 1]
```diff
@@ -49,33 +51,31 @@ def post_create_log_msg(appeal_submission_id:, submitted_appeal_uuid:)
...
     end
 
-    def handle_4142(request_body:, form4142:, response:, appeal_submission_id:, submitted_appeal_uuid:)
-      decision_review_service.process_form4142_submission(
-        request_body:, form4142:, user: @current_user, response:
-      )
-    rescue => e
-      handle_form4142_error(e, appeal_submission_id, submitted_appeal_uuid)
+    def handle_4142(request_body:, form4142:, appeal_submission_id:, submitted_appeal_uuid:)
+      rejiggered_payload = get_and_rejigger_required_info(request_body:, form4142:, user: @current_user)
```
Comment (rmtolmach) [inline app/controllers/v1/supplemental_claims_controller.rb:55]: 😆

**Review by rmtolmach — COMMENTED** [review]
Reviewed. @ me after you've gotten someone from your team to review and I can come back and approve if it's needed.

**Review by data-doge — APPROVED** [review]
LGTM!

**Comment by kylesoskin** [comment]
@rmtolmach can you re-approve when you get a chance? Thanks!

**Review by rmtolmach — APPROVED** [review]

**Pull request merged**